contracts/tokenfactory-issuer/src/msg.rs

```diff
@@ -2,8 +2,14 @@ use cosmwasm_schema::{cw_serde, QueryResponses};
 use cosmwasm_std::{Coin, Uint128};
 
 #[cw_serde]
-pub struct InstantiateMsg {
-    pub denom: String,
+pub enum InstantiateMsg {
```

Comment:
Woah, I didn't know InstantiateMsgs could be enums.  I tried to do this earlier but got annoyed with all the if statements and how ugly it got

Reply:
Yeah it has no rule as long as it's deserializable lol